cupy/cuda/memory.pyx

```diff
@@ -701,171 +805,103 @@ cdef class SingleDeviceMemoryPool:
         """
         return &self._index[stream_ptr]
 
-    cpdef _append_to_free_list(self, Py_ssize_t size, chunk,
-                               size_t stream_ptr):
-        cdef int index, bin_index
-        cdef list arena
-        cdef set free_list
-        cdef vector.vector[int]* arena_index
-
-        bin_index = self._bin_index_from_size(size)
-        rlock.lock_fastrlock(self._free_lock, -1, True)
-        try:
-            arena = self._arena(stream_ptr)
-            arena_index = self._arena_index(stream_ptr)
-            index = algorithm.lower_bound(
-                arena_index.begin(), arena_index.end(),
-                bin_index) - arena_index.begin()
-            size = <int>arena_index.size()
-            if index < size and arena_index.at(index) == bin_index:
-                free_list = arena[index]
-                if free_list is None:
-                    arena[index] = free_list = set()
-            else:
-                free_list = set()
-                arena_index.insert(arena_index.begin() + index, bin_index)
-                arena.insert(index, free_list)
-            free_list.add(chunk)
-        finally:
-            rlock.unlock_fastrlock(self._free_lock)
-
-    cpdef bint _remove_from_free_list(self, Py_ssize_t size, chunk,
-                                      size_t stream_ptr) except *:
-        """Removes the chunk from the free list.
+    cdef vector.vector[int8_t]* _arena_count(self, size_t stream_ptr):
```

Comment:
This function looks always returning 0 or 1. Is my understanding correct? (just want to confirm my understanding)

Reply:
No. This function is a No. This function is a same as `_arena` and `_arena_index`.same as `_arena` and `_arena_index`.